pkg/neorpc/rpcevent/filter.go

```diff
@@ -34,6 +34,13 @@ func Matches(f Comparator, r Container) bool {
 		return true
 	}
 	switch f.EventID() {
+	case neorpc.HeaderEventID:
+		filt := filter.(neorpc.BlockFilter)
+		b := r.EventPayload().(*block.Header)
+		primaryOk := filt.Primary == nil || *filt.Primary == int(b.PrimaryIndex)
+		sinceOk := filt.Since == nil || *filt.Since <= b.Index
+		tillOk := filt.Till == nil || b.Index <= *filt.Till
+		return primaryOk && sinceOk && tillOk
```

Comment:
DRY, unify this case with `case neorpc.BlockEventID` to avoid filter checks duplication.